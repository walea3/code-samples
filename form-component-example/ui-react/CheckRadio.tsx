import cx from 'classnames';
import React, { ReactNode } from 'react';

import { getDataRef } from '@nel-ui/js-helpers';
import { CheckRadioIcon } from './CheckRadioIcon';
import FormStyles from './FormStyles';

// TODO: Remove inline here when support is added
export type CheckRadioSingular = Omit<CheckRadioProps, 'inline' | 'variant'>;

export interface CheckRadioProps extends React.InputHTMLAttributes<HTMLInputElement> {
  id: string;
  label: ReactNode;
  variant: 'checkbox' | 'radio';
  alert?: 'warning' | 'error';
  inline?: boolean;
  disabled?: boolean;
  'data-ref'?: string;
}

export function CheckRadio({
  'data-ref': dataRef,
  variant,
  label,
  alert,
  inline,
  className,
  ...props
}: CheckRadioProps) {
  return (
    <FormStyles>
      {({ classes }) => (
        <div
          className={cx(classes.checkRadioContainer, {
            [classes.inlineCheckRadio]: inline,
          })}
          data-ref={getDataRef(dataRef, `${variant}Container`)}
        >
          <input
            data-ref={getDataRef(dataRef, `${variant}Input`)}
            type={variant}
            className={classes.checkRadioInput}
            aria-invalid={!!alert}
            {...props}
          />

          <label
            data-ref={getDataRef(dataRef, variant)}
            htmlFor={props.id}
            className={cx(className, classes.checkRadio, classes[variant], {
              [classes[`${alert}CheckRadio`]]: alert,
            })}
          >
            <span
              data-ref={getDataRef(dataRef, `${variant}Control`)}
              className={classes.checkRadioControl}
            >
              <CheckRadioIcon variant={variant} />
            </span>
            {label}
          </label>
        </div>
      )}
    </FormStyles>
  );
}
