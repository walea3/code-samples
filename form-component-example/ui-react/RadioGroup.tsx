import cx from 'classnames';
import React, { ReactText, useState } from 'react';

import { CheckRadioSingular, getIndexByValue, Radio } from './';
import FormStyles from './FormStyles';

type InputEvent = React.ChangeEvent<HTMLInputElement>;
type DivAttr = Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'>;

type RadioGroupItem = CheckRadioSingular & {
  value: ReactText;
};

export interface RadioGroupProps extends DivAttr {
  inline?: boolean;
  name: string;
  value?: RadioGroupItem['value'];
  items: RadioGroupItem[];
  'data-ref'?: string;
  onChange(
    selectedEvent: InputEvent,
    selected: {
      selectedItem: CheckRadioSingular;
      selectedIndex: number;
    }
  ): void;
}

export function RadioGroup({
  inline = false,
  className,
  items,
  onChange,
  'data-ref': dataRef,
  name,
  value,
  ...props
}: RadioGroupProps) {
  if (!items || (items && !items.length)) return null;

  const initialIndex = getIndexByValue(items, value);
  const [selectedItem, setSelected] = useState<number>(initialIndex);

  const handleChange = (index: number) => (event: InputEvent): void => {
    event.stopPropagation();

    setSelected(index);
    onChange(event, {
      selectedItem: items[index],
      selectedIndex: index,
    });
  };

  const displayMobileInline = items.length <= 2;

  return (
    <FormStyles>
      {({ classes }) => (
        <div className={cx(classes.checkRadioGroup, className)} {...props}>
          <div
            className={cx(classes.checkRadioGroupInner, {
              [classes.largeInlineGroup]: inline && !displayMobileInline,
              [classes.mobileInlineGroup]: inline && displayMobileInline,
            })}
          >
            {items.map(({ checked, ...radio }, i) => (
              <Radio
                key={radio.id}
                data-ref={dataRef ? `${dataRef}.${i}` : undefined}
                {...radio}
                name={name}
                inline={inline}
                className={classes.radioGroupItem}
                checked={i === selectedItem}
                onChange={handleChange(i)}
              />
            ))}
          </div>
        </div>
      )}
    </FormStyles>
  );
}
