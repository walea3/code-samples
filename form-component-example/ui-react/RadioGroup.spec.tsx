import { mountWithNEL, renderWithNEL } from '@nel-ui/tooling-react/test-utils';
import React from 'react';

import { clickRadio } from './CheckRadio.test-utils';
import { RadioGroup, RadioGroupProps } from './RadioGroup';

describe('<RadioGroup />', () => {
  const radioStubData = (valueIsString: boolean) =>
    Array.from({ length: 3 }, (_, i) => ({
      id: `radio-test-${i}`,
      value: valueIsString ? `radio-test-${i}` : i,
      label: `Item ${i}`,
    }));

  const requiredProps: RadioGroupProps = {
    name: 'test-list',
    items: radioStubData(true),
    onChange: jest.fn(),
  };

  const simulatedEventProps = {
    stopPropagation: jest.fn(),
  };

  beforeEach(jest.clearAllMocks);

  const renderComponent = (props: Partial<RadioGroupProps> = {}) =>
    renderWithNEL(<RadioGroup {...requiredProps} {...props} />);
  const mountComponent = (props: Partial<RadioGroupProps> = {}) =>
    mountWithNEL(<RadioGroup {...requiredProps} {...props} />);

  describe('@renders', () => {
    test.each`
      name                            | props
      ${'as default'}                 | ${{}}
      ${'with items value as number'} | ${{ items: radioStubData(false) }}
      ${'with no items'}              | ${{ items: [] }}
      ${'with inline'}                | ${{ inline: true }}
      ${'with className'}             | ${{ className: 'custom-className' }}
      ${'with value'}                 | ${{ value: 'radio-test-0' }}
      ${'with data-ref'}              | ${{ 'data-ref': 'uid' }}
      ${'with other props'}           | ${{ 'data-custom': 'uid' }}
    `('$name', ({ props }) => {
      expect(renderComponent(props)).toMatchSnapshot();
    });
  });

  describe('@events', () => {
    beforeEach(() => {
      expect(requiredProps.onChange).toHaveBeenCalledTimes(0);
    });

    describe('should stopPropagation on event', () => {
      beforeEach(() => {
        expect(simulatedEventProps.stopPropagation).toHaveBeenCalledTimes(0);
      });

      test.each`
        name                | selectOrder
        ${'single item'}    | ${[0]}
        ${'multiple items'} | ${[1, 2]}
      `('$name', ({ selectOrder }) => {
        const { root, wrapper } = mountComponent();

        selectOrder.forEach((checkboxIndex) =>
          clickRadio(wrapper, checkboxIndex, simulatedEventProps)
        );

        root.unmount();

        expect(simulatedEventProps.stopPropagation).toHaveBeenCalledTimes(
          selectOrder.length
        );
      });
    });

    describe('calls onChange()', () => {
      test.each`
        name                                                    | selectOrder
        ${'should select first item when none are preselected'} | ${[0]}
        ${'should return an updated radio selection'}           | ${[0, 2]}
      `('$name', ({ selectOrder }) => {
        const { root, wrapper } = mountComponent();

        const selectedIndex = selectOrder[selectOrder.length - 1];

        selectOrder.forEach((radioIndex) => clickRadio(wrapper, radioIndex));

        expect(requiredProps.onChange).toHaveBeenCalledTimes(selectOrder.length);
        expect(requiredProps.onChange).toHaveBeenCalledWith(
          expect.objectContaining({ type: 'change' }),
          {
            selectedItem: requiredProps.items[selectedIndex],
            selectedIndex,
          }
        );

        root.unmount();
      });
    });
  });
});
