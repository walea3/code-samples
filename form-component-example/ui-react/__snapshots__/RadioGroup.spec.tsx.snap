// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup /> @renders as default 1`] = `
<div
  class="nel-checkRadioGroup"
>
  <div
    class="nel-checkRadioGroupInner"
  >
    <div
      class="nel-checkRadioContainer"
      data-ref="radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="radioInput"
        id="radio-test-0"
        name="test-list"
        type="radio"
        value="radio-test-0"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="radio"
        for="radio-test-0"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 0
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="radioInput"
        id="radio-test-1"
        name="test-list"
        type="radio"
        value="radio-test-1"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="radio"
        for="radio-test-1"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 1
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="radioInput"
        id="radio-test-2"
        name="test-list"
        type="radio"
        value="radio-test-2"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="radio"
        for="radio-test-2"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 2
      </label>
    </div>
  </div>
</div>
`;

exports[`<RadioGroup /> @renders with className 1`] = `
<div
  class="nel-checkRadioGroup custom-className"
>
  <div
    class="nel-checkRadioGroupInner"
  >
    <div
      class="nel-checkRadioContainer"
      data-ref="radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="radioInput"
        id="radio-test-0"
        name="test-list"
        type="radio"
        value="radio-test-0"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="radio"
        for="radio-test-0"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 0
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="radioInput"
        id="radio-test-1"
        name="test-list"
        type="radio"
        value="radio-test-1"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="radio"
        for="radio-test-1"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 1
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="radioInput"
        id="radio-test-2"
        name="test-list"
        type="radio"
        value="radio-test-2"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="radio"
        for="radio-test-2"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 2
      </label>
    </div>
  </div>
</div>
`;

exports[`<RadioGroup /> @renders with data-ref 1`] = `
<div
  class="nel-checkRadioGroup"
>
  <div
    class="nel-checkRadioGroupInner"
  >
    <div
      class="nel-checkRadioContainer"
      data-ref="uid.0.radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="uid.0.radioInput"
        id="radio-test-0"
        name="test-list"
        type="radio"
        value="radio-test-0"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="uid.0.radio"
        for="radio-test-0"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="uid.0.radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 0
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="uid.1.radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="uid.1.radioInput"
        id="radio-test-1"
        name="test-list"
        type="radio"
        value="radio-test-1"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="uid.1.radio"
        for="radio-test-1"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="uid.1.radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 1
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="uid.2.radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="uid.2.radioInput"
        id="radio-test-2"
        name="test-list"
        type="radio"
        value="radio-test-2"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="uid.2.radio"
        for="radio-test-2"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="uid.2.radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 2
      </label>
    </div>
  </div>
</div>
`;

exports[`<RadioGroup /> @renders with inline 1`] = `
<div
  class="nel-checkRadioGroup"
>
  <div
    class="nel-checkRadioGroupInner nel-largeInlineGroup"
  >
    <div
      class="nel-checkRadioContainer nel-inlineCheckRadio"
      data-ref="radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="radioInput"
        id="radio-test-0"
        name="test-list"
        type="radio"
        value="radio-test-0"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="radio"
        for="radio-test-0"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 0
      </label>
    </div>
    <div
      class="nel-checkRadioContainer nel-inlineCheckRadio"
      data-ref="radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="radioInput"
        id="radio-test-1"
        name="test-list"
        type="radio"
        value="radio-test-1"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="radio"
        for="radio-test-1"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 1
      </label>
    </div>
    <div
      class="nel-checkRadioContainer nel-inlineCheckRadio"
      data-ref="radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="radioInput"
        id="radio-test-2"
        name="test-list"
        type="radio"
        value="radio-test-2"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="radio"
        for="radio-test-2"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 2
      </label>
    </div>
  </div>
</div>
`;

exports[`<RadioGroup /> @renders with items value as number 1`] = `
<div
  class="nel-checkRadioGroup"
>
  <div
    class="nel-checkRadioGroupInner"
  >
    <div
      class="nel-checkRadioContainer"
      data-ref="radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="radioInput"
        id="radio-test-0"
        name="test-list"
        type="radio"
        value="0"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="radio"
        for="radio-test-0"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 0
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="radioInput"
        id="radio-test-1"
        name="test-list"
        type="radio"
        value="1"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="radio"
        for="radio-test-1"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 1
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="radioInput"
        id="radio-test-2"
        name="test-list"
        type="radio"
        value="2"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="radio"
        for="radio-test-2"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 2
      </label>
    </div>
  </div>
</div>
`;

exports[`<RadioGroup /> @renders with no items 1`] = `null`;

exports[`<RadioGroup /> @renders with other props 1`] = `
<div
  class="nel-checkRadioGroup"
  data-custom="uid"
>
  <div
    class="nel-checkRadioGroupInner"
  >
    <div
      class="nel-checkRadioContainer"
      data-ref="radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="radioInput"
        id="radio-test-0"
        name="test-list"
        type="radio"
        value="radio-test-0"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="radio"
        for="radio-test-0"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 0
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="radioInput"
        id="radio-test-1"
        name="test-list"
        type="radio"
        value="radio-test-1"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="radio"
        for="radio-test-1"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 1
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="radioInput"
        id="radio-test-2"
        name="test-list"
        type="radio"
        value="radio-test-2"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="radio"
        for="radio-test-2"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 2
      </label>
    </div>
  </div>
</div>
`;

exports[`<RadioGroup /> @renders with value 1`] = `
<div
  class="nel-checkRadioGroup"
>
  <div
    class="nel-checkRadioGroupInner"
  >
    <div
      class="nel-checkRadioContainer"
      data-ref="radioContainer"
    >
      <input
        aria-invalid="false"
        checked=""
        class="nel-checkRadioInput"
        data-ref="radioInput"
        id="radio-test-0"
        name="test-list"
        type="radio"
        value="radio-test-0"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="radio"
        for="radio-test-0"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 0
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="radioInput"
        id="radio-test-1"
        name="test-list"
        type="radio"
        value="radio-test-1"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="radio"
        for="radio-test-1"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 1
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="radioContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="radioInput"
        id="radio-test-2"
        name="test-list"
        type="radio"
        value="radio-test-2"
      />
      <label
        class="nel-checkRadio nel-radio"
        data-ref="radio"
        for="radio-test-2"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="radioControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
            />
          </svg>
        </span>
        Item 2
      </label>
    </div>
  </div>
</div>
`;
